docs/FlashAndTheBug.md

```diff
@@ -153,3 +155,21 @@ Open a second terminal session to use GDB
 3. Run `tar extended-remote :3333` to connect to the OpenOCD GDB server.
 
 Step through your code in GDB to analyze execution!
+
+#### Serial Monitoring
+
+Another debugging option is serial monitoring. The `printf` method is integrated into Embedded-Sharepoint.
+
+1. Run `printf_init(UART_HandleTypeDef)` with your desired UART to output to. For a Nucleo, this will be specified in the Nucleo user manual which you can find online. For one of our PCBs, check the schematic to see which UART peripheral your USB is connected to. 
+
+- `printf_init` must be run after the RTOS is initialized.
+- the `UART_HandleTypeDef` struct must be initialized with the proper settings before calling printf_init.
```

Comment:
This information should come before telling people to call `printf_init(UART_HandleTypeDef*)`